src/lib/components/Swap/Toolbar.tsx

```diff
@@ -22,30 +29,29 @@ function RoutingTooltip() {
   )
 }
 
-type FilledInput = Input & Required<Pick<Input, 'token' | 'value'>>
-
-function asFilledInput(input: Input): FilledInput | undefined {
-  return input.token && input.value ? (input as FilledInput) : undefined
-}
-
 interface LoadedStateProps {
-  input: FilledInput
-  output: FilledInput
+  inputAmount: CurrencyAmount<Currency>
+  outputAmount: CurrencyAmount<Currency>
+  trade: InterfaceTrade<Currency, Currency, TradeType> | undefined
 }
 
-function LoadedState({ input, output }: LoadedStateProps) {
+function LoadedState({ inputAmount, outputAmount, trade }: LoadedStateProps) {
   const [flip, setFlip] = useState(true)
+  const executionPrice = trade?.executionPrice
+  const usdcValue = useUSDCPrice(inputAmount.currency)
+
   const ratio = useMemo(() => {
-    const [a, b] = flip ? [output, input] : [input, output]
-    const ratio = `1 ${a.token.symbol} = ${b.value / a.value} ${b.token.symbol}`
-    const usdc = a.usdc && ` ($${(a.usdc / a.value).toLocaleString('en')})`
+    const [a, b] = flip ? [outputAmount, inputAmount] : [inputAmount, outputAmount]
+
+    const ratio = `1 ${a.currency.symbol} = ${executionPrice?.toSignificant(6)} ${b.currency.symbol}`
+    const usdc = usdcValue ? ` ($${usdcValue.toSignificant(2)})` : null
```

Comment:
USDC should refer to the latter CurrencyAmount, so it'll need to be fetched for both input and output amounts (see old code).